src/league-of-legends-api/poll-current-game.ts

```diff
@@ -32,173 +26,187 @@ export const getAllGameData = async () => {
         httpsAgent
       }
     );
-    setCachedGame(rootGameObject);
+    cachedGame = rootGameObject;
   } catch (err: unknown | AxiosError) {
     const ERROR_MSG = `Error occured when getting game data: ${err}`;
     console.log(ERROR_MSG);
   }
 };
 
-export const pollCurrentGame = (
+export const startPollingLoLGame = (
   channel: VoiceBasedChannel,
   audioPlayer: AudioPlayer,
   pathToClips: string
-): NodeJS.Timer => {
-  return setIntervalImmediately(async () => {
-    try {
-      if (!cachedGame) {
-        getAllGameData();
-      }
-      const {
-        data: { Events: currentEvents }
-      }: AxiosResponse<RootEventsObject> = await axios.get(`${LOL_GAME_CLIENT_API}/eventdata`, {
-        httpsAgent
-      });
-      if (currentEvents.length > cachedEvents.length) {
-        const newEvent = currentEvents[currentEvents.length - 1];
-        switch (newEvent?.EventName) {
-          case LoLClientEvent.GAME_START: {
-            console.log('game start!');
-            await playClip(`${pathToClips}halo_slayer.mp3`, channel, audioPlayer);
-            break;
-          }
-          case LoLClientEvent.FIRST_BLOOD: {
-            console.log('first blood!');
-            await playClip(`${pathToClips}firstblood.mp3`, channel, audioPlayer);
-            break;
-          }
-          case LoLClientEvent.CHAMPION_KILL: {
-            if (cachedGame) {
-              // Determine who is on the active player's team
-              const activePlayerSummonerName = cachedGame.activePlayer.summonerName;
-              const activePlayerTeam = cachedGame.allPlayers.find(
-                (p) => p.summonerName === activePlayerSummonerName
-              )?.team;
-              const victimSummonerName = newEvent?.VictimName;
-              const victimTeam = cachedGame.allPlayers.find(
-                (p) => p.summonerName === victimSummonerName
-              )?.team;
+) => {
+  if (leagueOfLegendsPollTimer === null) {
```

Comment:
Same functionality in this function here, just everything wrapped in this `if` statement.